kubeconfig/src/Kubernetes/KubeConfig.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+
+{-|
+Module      : Kubernetes.KubeConfig
+Description : Data model for the kubeconfig.
+
+This module contains the definition of the data model of the kubeconfig.
+
+The official definition of the kubeconfig is defined in https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/api/v1/types.go.
+
+This is a mostly straightforward translation into Haskell, with 'FromJSON' and 'ToJSON' instances defined.
+-}
+module Kubernetes.KubeConfig where
+
+import           Data.Aeson       (FromJSON (..), Options, ToJSON (..),
+                                   Value (..), camelTo2, defaultOptions,
+                                   fieldLabelModifier, genericParseJSON,
+                                   genericToJSON, object, omitNothingFields,
+                                   withObject, (.:), (.=))
+import qualified Data.Map         as Map
+import           Data.Proxy
+import           Data.Semigroup   ((<>))
+import           Data.Text        (Text)
+import qualified Data.Text        as T
+import           Data.Typeable
+import           GHC.Generics
+import           GHC.TypeLits
+
+camelToWithOverrides :: Char -> Map.Map String String -> Options
+camelToWithOverrides c overrides = defaultOptions
+    { fieldLabelModifier = modifier
+    , omitNothingFields  = True
+    }
+    where modifier s = Map.findWithDefault (camelTo2 c s) s overrides
+
+-- |Represents a kubeconfig.
+data Config = Config
+  { kind           :: Maybe (Text)
```

Comment:
You don't need parentheses here when the type parameter is just a type name.

Reply:
Done